### PR TITLE
Port histogram from jaspDescriptives to jaspGraphs

### PR DESCRIPTION
**Pull request by vandenman**
Per request of @sophieberkhout.

In addition, some modules import jaspDescriptives purely for the histogram (such as jaspProcessControl). By putting the function inside jaspGraphs we avoid that dependency and have jaspGraphs be the common denominator for plots (on which the modules depend anyway).

man/geom_abline2.Rd was auto-generated by roxygen2.

See https://github.com/jasp-stats/jaspDescriptives/blob/667a2f78c9ffcbcc79725e91f96144e66e63f7d6/R/descriptives.R#L1148-L1240 for the implementation in jaspDescriptives.

Don't merge before 0.16.2!

### CODE REVIEW

File: R/jaspHistogram.R [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+  if (binWidthType == "doane") {
+
+    # https://en.wikipedia.org/wiki/Histogram#Doane's_formula
+    sigma.g1 <- sqrt((6*(length(x) - 2)) / ((length(x) + 1)*(length(x) + 3)))
+    g1 <- mean(abs(x)^3)
+    k <- 1 + log2(length(x)) + log2(1 + (g1 / sigma.g1))
+    binWidthType <- k
+
+  } else if (binWidthType == "fd" && grDevices::nclass.FD(x) > 10000) { # FD-method will produce extreme number of bins and crash ggplot, mention this in footnote
```
Comment (Kucharssim) [inline R/jaspHistogram.R:31]:
```suggestion
  } else if (binWidthType == "fd" && grDevices::nclass.FD(x) > 10000) { # FD-method will produce extreme number of bins and crash ggplot, mention this in footnote
  warning("The Freedman-Diaconis method would produce an extreme number of bins, setting the number of bins to 10,000.")
```

**Review by Kucharssim — CHANGES_REQUESTED** [review]
Looks good, I suggested some minor changes.

It seems to me that @arandolan will probably need to adapt to this change as he worked on extending the histogram functionality in jaspDescriptives. Aran, I will talk to you about this.

File: R/jaspHistogram.R [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+    binWidthType <- 10000
+
+  } else if (binWidthType == "manual") {
```
Comment (Kucharssim) [inline R/jaspHistogram.R:35]:
```suggestion
  } else if (binWidthType == "manual") {
  if(is.na(numberOfBins)) stop("numberOfBins argument must be specified when a binWidthType == 'manual'.")
```
Just fail early with an informative message

**Comment by Kucharssim** [comment]
> Don't merge before 0.16.2!

Technically merging this should be harmless, right? Since the new jaspGraphs function is not yet used anywhere in JASP. It could help @arandolan or @sophieberkhout use this functionality for development without needing to checkout your branch. But if 0.16.2 rolls out relatively fast it's probably not a big deal.

**Comment by vandenman** [comment]
> Technically merging this should be harmless, right?

Well, if this does get merged and for some reason, we need to update the R-framework, we need to do some annoying changes (e.g., add commit hashes to all jasp-stats/jaspGraphs in the DESCRIPTION files).

File: R/jaspHistogram.R [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+  density = FALSE,
+  densityColor = FALSE,
+  densityShade = FALSE,
+  densityShadeAlpha = 0.6,
+  densityLineWidth = 1,
+  hideYAxisLabels = density) {
+
+  # validate input
+  if (!is.vector(x, mode = "numeric"))
+    stop2("`x` must be a numeric vector but has class ")
```
Comment (Kucharssim) [inline R/jaspHistogram.R:47]:
```suggestion
    stop2("`x` must be a numeric vector but has class ", paste(class(x), collapse=", "))
```
?

File: R/jaspHistogram.R [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+
+  if (missing(xName))
+    xName <- deparse1(substitute(x)) # identical to plot.default
```
Comment (Kucharssim) [inline R/jaspHistogram.R:50]: since you do this for `xName` (nice!), you could also do this for `groupingVariableName`.

**Review by Kucharssim — APPROVED** [review]
Really nice! The comments are cosmetic, it's up to you whether you want to change it.

**Pull request merged**